databuilder/dsl.py

```diff
@@ -367,6 +413,12 @@ def convert_to_days(self):
     def convert_to_weeks(self):
         return self._convert("weeks")
 
+    def __rsub__(self, other: str) -> DateDeltaSeries:
+        datestring = _validate_datestring(other)
+        # This allows subtraction of a DateDeltaSeries from a date string
+        # e.g. 2020-10-01
+        return DateDeltaSeries(DateSubtraction(datestring, self.convert_to_days()))
```

Comment:
Note this currently only allows you to do things like `"2020-10-01" - <DateDeltaSeries>`.  I'm leaving adding/subtracting int/DateDeltaSeries to/from a DateDeltaSeries for another PR

Reply:
The rest of the subtraction is in this PR after all now